src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2675,6 +2672,40 @@ def convert_func_to_agg_func_info(
         new_data_column_quoted_identifier = [
             col_agg_op.agg_snowflake_quoted_identifier for col_agg_op in agg_col_ops
         ]
+
+        # The ordering of the named aggregations is changed by us when we process
+        # the agg_kwargs into the func dict (named aggregations on the same
+        # column are moved to be contiguous, see groupby.py::aggregate for an
+        # example). We need to check if the order of the output columns is correct,
+        # and if not, reorder them.
+        if uses_named_aggs:
+            correct_ordering = list(agg_kwargs.keys())
+            if correct_ordering != new_data_column_pandas_labels:
+                # In this case, we need to reorder the new_data_column_pandas_labels
+                # and the new_data_column_quoted_identifier.
+                data_column_label_to_quoted_identifier = list(
+                    zip(
+                        new_data_column_pandas_labels, new_data_column_quoted_identifier
+                    )
+                )
+                data_column_label_to_quoted_identifier = list(
+                    zip(
+                        *[
+                            pair
+                            for column_label in correct_ordering
+                            for pair in filter(
+                                lambda pair: pair[0] == column_label,
+                                data_column_label_to_quoted_identifier,
+                            )
+                        ]
+                    )
+                )
+                new_data_column_pandas_labels = list(
+                    data_column_label_to_quoted_identifier[0]
+                )
+                new_data_column_quoted_identifier = list(
+                    data_column_label_to_quoted_identifier[1]
+                )
```

Comment:
```suggestion
                new_data_column_pandas_labels, new_data_column_quoted_identifier = list(
                    zip(
                        *[
                            pair
                            for column_label in correct_ordering
                            for pair in filter(
                                lambda pair: pair[0] == column_label,
                                data_column_label_to_quoted_identifier,
                            )
                        ]
                    )
                )
```

IMO tuple assignment is easier to read.

You could also consider directly sorting `data_column_label_to_quoted_identifier` by key= each pair's index in `correct_ordering`, but I'm not sure if that would make the code harder to read or not.